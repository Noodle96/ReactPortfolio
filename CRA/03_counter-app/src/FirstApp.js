import { Fragment } from "react";
import PropTypes from "prop-types";

//FUNCTIONAL COMPONENTS
const FirstApp = ({saludo, subtitles}) => {
    const card = 'Princess';
    const student = {
        name : 'Pedro',
        age: 34,
    }
    const heigh = 34.677;

    // console.log(saludo);
    // console.log(subtitles);

    // if(!saludo){
    //     throw new Error('el saludo es necesario');
    // }


    return (
        <Fragment>
            <h1>{saludo}</h1>
            <h1 id='subtitle'>{subtitles}</h1>
            {/* <h1>Hello world with functional components</h1>
            <p> the card is: {card} </p>
            <pre> {JSON.stringify(student,null,4)}</pre>
            <h1>{heigh}</h1> */}
        </Fragment>
    );
}

FirstApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitles: 'template subtitulo'
}

export default FirstApp;