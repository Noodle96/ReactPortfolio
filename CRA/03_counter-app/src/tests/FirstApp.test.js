
import React from 'react'
// import {render} from "@testing-library/react"
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import FirstApp from "../FirstApp";

describe('Pruebas en FirstApp', () => {
    // test('Deberia mostrar el mensaje "RUSSELL AI"', () => { 
    //     const saludo_evaluar = 'RUSSELL AI';
    //     //producto renderizado de lo quiero mostrar
    //     const {getByText} = render(<FirstApp saludo={saludo_evaluar}/>);
    //     expect(getByText(saludo_evaluar)).toBeInTheDocument();
    //  })

    test('Deberia mostrar <FirstApp/> correctamente', () => { 
        const saludo_evaluar = 'RUSSELL AI';
        const wrapper = shallow(<FirstApp saludo={saludo_evaluar}/>);
        // console.log(wrapper);
        expect(wrapper).toMatchSnapshot();
     })

     test('Deberia de mostrar el subtitulo enviado por props', () =>  {
        const saludo = 'Welcome to enzyme testing';
        const subtitulo = 'subtitle in enzyme';
        const wrapper = shallow(
            <FirstApp
                 saludo={saludo}
                 subtitles={subtitulo}
            />
        );
        const textoSubtitulo = wrapper.find('#subtitle').text();
        // console.log( textoSubtitulo);
        expect(textoSubtitulo).toBe(subtitulo);

     });
});
