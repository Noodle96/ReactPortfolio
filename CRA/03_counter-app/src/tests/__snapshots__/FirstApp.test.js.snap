// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Deberia mostrar <FirstApp/> correctamente 1`] = `
<Fragment>
  <h1>
    RUSSELL AI
  </h1>
  <h1
    id="subtitle"
  >
    template subtitulo
  </h1>
</Fragment>
`;
